src/main/java/com/minecolonies/coremod/commands/CommandEntryPointNew.java

```diff
@@ -442,13 +465,18 @@ private ParsingResult getTabCompletionsAndParsingHolders(
             final List<ActionMenuHolder> parsedHolders = new ArrayList<>();
             final Map<String, ActionMenuHolder> possibleActionCommands = getActionCommands(treeNode, parsedHolders);
             @NotNull final List<ActionArgument> parsedActionArgumentList = new ArrayList<>();
-            return getTabCompletionsAndParsingHoldersForActionMenuTreeNode(treeNode, parsedHolders, parsedActionArgumentList, possibleActionCommands, server, sender, args, pos);
+            final ActionMenu actionMenu = (ActionMenu) treeNode.getData();
+            @NotNull final ActionMenuState actionMenuState = new ActionMenuState(actionMenu);
+            return getTabCompletionsAndParsingHoldersForActionMenuTreeNode(treeNode, actionMenuState, parsedHolders, parsedActionArgumentList, possibleActionCommands, server,
+                    sender, args, pos);
         }
     }
 
     @NotNull
     private ParsingResult getTabCompletionsAndParsingHoldersForActionMenuTreeNode(
                                            @NotNull final TreeNode<IMenu> actionMenuTreeNode,
+           // TODO: can parsedHolders, parsedActionArgumentList, and possibleActionCommands be merged into actionMenuState?
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)